packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -513,6 +520,36 @@ class Treemap extends StatelessWidget {
       },
     );
   }
+
+  /// Checks if the touch point of the given [details] is overlapping with
+  /// a cell in [positionedCells].
+  /// 
+  /// If so, reroot to the matching cell if the cell has children to show.
+  void _onTapDown(
+    TapDownDetails details,
+    List<PositionedCell> positionedCells,
+  ) {
+    final x = details.localPosition.dx;
+    final y = details.localPosition.dy;
+    // Define target point as rectangle with width and height as 0.01 to
+    // represent the touch point.
+    final touchPoint = Offset(x, y);
+    // TODO(peterdjlee): Optimize with better algorithm to find the overlapping cell.
+    for (final cell in positionedCells) {
+      final cellRect = Rect.fromLTWH(
+        cell.left,
```

Comment:
we should just store these values in a rect on the cell instead of having individual fields for left, top, width, height on the cell

Reply:
Done